geomstats/geometry/hypersphere.py

```diff
@@ -737,6 +737,98 @@ def curvature(
         second_term = gs.einsum('...,...i->...i', inner_ac, tangent_vec_b)
         return - first_term + second_term
 
+    def _normalization_factor_odd_dim(self, variances):
+        """Compute the normalization factor - odd dimension."""
+        dim = self.dim
+        half_dim = int((dim + 1) / 2)
+        area = 2 * gs.pi ** half_dim / math.factorial(half_dim - 1)
+        comb = math.comb(dim - 1, half_dim - 1)
+
+        erf_arg = gs.sqrt(variances / 2) * gs.pi
+        first_term = area / (2 ** dim - 1) * comb * gs.sqrt(
+            gs.pi / (2 * variances)) * gs.erf(erf_arg)
+
+        def summand(k):
+            exp_arg = - (dim - 1 - 2 * k) ** 2 / 2 / variances
+            erf_arg_2 = (gs.pi * variances - (dim - 1 - 2 * k) * 1j) / gs.sqrt(
+                2 * variances)
+            sign = (- 1.) ** k
+            comb_2 = math.comb(k, dim - 1)
+            return sign * comb_2 * gs.exp(exp_arg) * gs.real(gs.erf(erf_arg_2))
+
+        if half_dim > 2:
+            sum_term = gs.sum(
+                gs.stack([summand(k)] for k in range(half_dim - 2)))
+        else:
+            sum_term = summand(0)
+        coef = area / 2 / erf_arg * gs.pi ** .5 * (- 1.) ** (half_dim - 1)
+
+        return first_term + coef / 2 ** (dim - 2) * sum_term
+
+    def _normalization_factor_even_dim(self, variances):
+        """Compute the normalization factor - even dimension."""
+        dim = self.dim
+        half_dim = (dim + 1) / 2
+        area = 2 * gs.pi ** half_dim / math.gamma(half_dim)
+
+        def summand(k):
+            exp_arg = - (dim - 1 - 2 * k) ** 2 / 2 / variances
+            erf_arg_1 = (dim - 1 - 2 * k) * 1j / gs.sqrt(2 * variances)
+            erf_arg_2 = (gs.pi * variances - (dim - 1 - 2 * k) * 1j) / gs.sqrt(
+                2 * variances)
+            sign = (- 1.) ** k
+            comb = math.comb(dim - 1, k)
+            erf_terms = gs.imag(gs.erf(erf_arg_2) + gs.erf(erf_arg_1))
+            return sign * comb * gs.exp(exp_arg) * erf_terms
+
+        half_dim_2 = int((dim - 2) / 2)
+        if half_dim_2 > 0:
+            sum_term = gs.sum(
+                gs.stack([summand(k)] for k in range(half_dim_2)))
+        else:
+            sum_term = summand(0)
+        coef = area * (- 1.) ** half_dim_2 / 2 ** (dim - 2) * gs.sqrt(
+            gs.pi / 2 / variances)
+
+        return coef * sum_term
+
+    def normalization_factor(self, variances):
```

Comment:
Maybe call it `norm_factor` everywhere, since this is what is used in the name of the gradient?

Reply:
But it's less explicit as it can be confused with the norm of the metric